per-user-usage.py

```diff
@@ -1,118 +1,230 @@
+        ContextData(),
+        ObjectType(ObjectIdentity(OID)),
+        # ObjectType(ObjectIdentity('1.3.6.1.2.1.31.1.1.1.6')), # (64 bit - but not supported by Cisco)
+        lexicographicMode=False
+        ):
 
         if thread_data.errorIndication:
             print(thread_data.errorIndication)
             break
         elif thread_data.errorStatus:
-            print('%s at %s' % (thread_data.errorStatus.prettyPrint(),
-                                thread_data.errorIndex and thread_data.varBinds[int(thread_data.errorIndex) - 1][0] or '?'))
+            print(
+                '%s at %s' % (thread_data.errorStatus.prettyPrint(),
+                              thread_data.errorIndex and thread_data.varBinds[int(thread_data.errorIndex) - 1][0] or '?')
+                )
             break
         else:
             for thread_data.varBind in thread_data.varBinds:
                 thread_data.uid = str(thread_data.varBind[0]).split(".")
                 try:
-                    interface_data[thread_data.uid[-1]] = int(thread_data.varBind[1])
+                    interface_data[str(thread_data.uid[-1])] = int(thread_data.varBind[1])
                 except KeyError:
                     # print("Interface ID "+uid[-1]+" not found")
                     pass
 
     print('Got ' + str(len(interface_data)) + ' data entries for OID: ' + OID)
 
+
+def convert_timetick_to_datetime(ticks):
+    """This function converts timetick to timedelta"""
+    seconds = ticks / 100
+    return_time_delta = (datetime.timedelta(seconds=seconds))
+    return return_time_delta
+
+
+def collect_session_data_from_lns(SNMP_target, auth_data, sessions_uptime, OID):
+    print('Getting stats for OID: ', OID, flush=True)
+    thread_data = threading.local()
+    for (thread_data.errorIndication,
+         thread_data.errorStatus,
+         thread_data.errorIndex,
+         thread_data.varBinds) in nextCmd(
+        SnmpEngine(),
+        auth_data,
+        SNMP_target,
+        ContextData(),
+        ObjectType(ObjectIdentity(OID)),
+        # ObjectType(ObjectIdentity('1.3.6.1.2.1.31.1.1.1.6')), # (64 bit - but not supported by Cisco)
+        lexicographicMode=False
+        ):
+
+        if thread_data.errorIndication:
+            print(thread_data.errorIndication)
+            break
+        elif thread_data.errorStatus:
+            print(
+                '%s at %s' % (thread_data.errorStatus.prettyPrint(),
+                              thread_data.errorIndex and thread_data.varBinds[int(thread_data.errorIndex) - 1][0] or '?')
+                )
+            break
+        else:
+            for thread_data.varBind in thread_data.varBinds:
+                thread_data.uid = str(thread_data.varBind[0]).split(".")
+                try:
+                    time_delta_string = convert_timetick_to_datetime(int(thread_data.varBind[1]))
+                    sessions_uptime[str(thread_data.uid[-2] + '.' + thread_data.uid[-1])] = str(time_delta_string)
```

Comment:
prometheus  can't collect texts.  only floats.

Reply:
resolved in the latest commit